crates/client/eth/src/state_update.rs

```diff
@@ -53,12 +55,12 @@ pub fn update_l1(
     backend: &MadaraBackend,
     state_update: L1StateUpdate,
     block_metrics: &L1BlockMetrics,
-    chain_id: Felt,
+    chain_id: ChainId,
 ) -> anyhow::Result<()> {
     // This is a provisory check to avoid updating the state with an L1StateUpdate that should not have been detected
     //
     // TODO: Remove this check when the L1StateUpdate is properly verified
-    if state_update.block_number > 500000u64 || chain_id == MAIN_CHAIN_ID {
+    if state_update.block_number > 500000u64 || chain_id.to_felt() == MAIN_CHAIN_ID {
```

Comment:
I don't really understand this logic and the 'or'

Reply:
I believe the logic was in place before this and if I remember correctly it was because there was some mismatch before specific on mainnet